vkcubepp (KHR_wayland_surface) fails to build since v1.3.209

Regressed by 3bcefe5770cf
```c++
$ cmake -DCUBE_WSI_SELECTION=WAYLAND -B /tmp/vulkan-tools_build
$ cmake --build /tmp/vulkan-tools_build
[...]
cube/cube.cpp:489:9: error: no matching function for call to 'wl_pointer_add_listener'
        wl_pointer_add_listener(demo.pointer, &pointer_listener, demo);
        ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/wayland-client-protocol.h:4671:1: note: candidate function not viable: no known conversion from 'Demo' to 'void *' for 3rd argument; take the address of the argument with &
wl_pointer_add_listener(struct wl_pointer *wl_pointer,
^
cube/cube.cpp:497:9: error: no matching function for call to 'wl_keyboard_add_listener'
        wl_keyboard_add_listener(demo.keyboard, &keyboard_listener, demo);
        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/wayland-client-protocol.h:4986:1: note: candidate function not viable: no known conversion from 'Demo' to 'void *' for 3rd argument; take the address of the argument with &
wl_keyboard_add_listener(struct wl_keyboard *wl_keyboard,
^
cube/cube.cpp:522:9: error: no matching function for call to 'wl_seat_add_listener'
        wl_seat_add_listener(demo.seat, &seat_listener, demo);
        ^~~~~~~~~~~~~~~~~~~~
/usr/include/wayland-client-protocol.h:4171:1: note: candidate function not viable: no known conversion from 'Demo' to 'void *' for 3rd argument; take the address of the argument with &
wl_seat_add_listener(struct wl_seat *wl_seat,
^
3 errors generated.
```

--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -486,7 +486,7 @@ static void seat_handle_capabilities(void *data, wl_seat *seat, uint32_t caps) {
     Demo &demo = *static_cast<Demo *>(data);
     if ((caps & WL_SEAT_CAPABILITY_POINTER) && !demo.pointer) {
         demo.pointer = wl_seat_get_pointer(seat);
-        wl_pointer_add_listener(demo.pointer, &pointer_listener, demo);
+        wl_pointer_add_listener(demo.pointer, &pointer_listener, &demo);
     } else if (!(caps & WL_SEAT_CAPABILITY_POINTER) && demo.pointer) {
         wl_pointer_destroy(demo.pointer);
         demo.pointer = nullptr;
@@ -494,7 +494,7 @@ static void seat_handle_capabilities(void *data, wl_seat *seat, uint32_t caps) {
     // Subscribe to keyboard events
     if (caps & WL_SEAT_CAPABILITY_KEYBOARD) {
         demo.keyboard = wl_seat_get_keyboard(seat);
-        wl_keyboard_add_listener(demo.keyboard, &keyboard_listener, demo);
+        wl_keyboard_add_listener(demo.keyboard, &keyboard_listener, &demo);
     } else if (!(caps & WL_SEAT_CAPABILITY_KEYBOARD)) {
         wl_keyboard_destroy(demo.keyboard);
         demo.keyboard = nullptr;
@@ -519,7 +519,7 @@ static void registry_handle_global(void *data, wl_registry *registry, uint32_t i
         xdg_wm_base_add_listener(demo.wm_base, &wm_base_listener, nullptr);
     } else if (strcmp(interface, wl_seat_interface.name) == 0) {
         demo.seat = (wl_seat *)wl_registry_bind(registry, id, &wl_seat_interface, 1);
-        wl_seat_add_listener(demo.seat, &seat_listener, demo);
+        wl_seat_add_listener(demo.seat, &seat_listener, &demo);
     } else if (strcmp(interface, zxdg_decoration_manager_v1_interface.name) == 0) {
         demo.xdg_decoration_mgr =
             (zxdg_decoration_manager_v1 *)wl_registry_bind(registry, id, &zxdg_decoration_manager_v1_interface, 1);

